proof-pile-v2/source_code/process_github.py

```diff
@@ -0,0 +1,331 @@
+import os
+import re
+import hashlib
+from pathlib import Path
+from github import Github
+from tqdm import tqdm
+from datetime import datetime
+import ndjson
+import tiktoken
+import json
+import random
+import numpy as np
+
+
+GITHUB_ACCESS_TOKEN = os.environ['GITHUB_ACCESS_TOKEN']
+
+TEXT_MAX_SIZE = 1048575  # in bytes
+MAX_NUMERICAL_DENSITY = .5
+
+
+def _get_dir_from_repo(author, repo, sha, repo_dir, save_path, overwrite):
```

Comment:
Also maybe worth adding exponential backoff in case of network problems/rate limits. Something like `import backoff; @backoff.on_exception(backoff.expo)`

Reply:
Done